app/jobs/fixed_asset_export_job.rb

```diff
@@ -0,0 +1,22 @@
+class FixedAssetExportJob < ActiveJob::Base
+  queue_as :default
+  include Rails.application.routes.url_helpers
+
+  def perform(document_nature, key, state, period, user)
+    begin
+      fixed_asset_printer = "#{state.split('_').map(&:capitalize).join}FixedAssetPrinter".constantize.new(document_nature: document_nature,
```

Comment:
Can we break this into a few disticnt variables?
```suggestion
      printer_class_name = "#{state.split('_').map(&:capitalize).join}FixedAssetPrinter"
      printer_class = printer_class_name.constantize
      fixed_asset_printer = printer_class.new(document_nature: document_nature,
```